src/storage/store/column.cpp

```diff
@@ -333,23 +371,25 @@ void Column::scan(transaction::Transaction* transaction, node_group_idx_t nodeGr
         nullColumn->scan(transaction, nodeGroupIdx, startOffsetInGroup, endOffsetInGroup,
             resultVector, offsetInVector);
     }
-    auto chunkMeta = metadataDA->get(nodeGroupIdx, transaction->getType());
-    auto pageCursor = PageUtils::getPageElementCursorForPos(startOffsetInGroup,
-        chunkMeta.compMeta.numValues(BufferPoolConstants::PAGE_4KB_SIZE, dataType));
-    pageCursor.pageIdx += chunkMeta.pageIdx;
+    auto state = getReadState(transaction->getType(), nodeGroupIdx);
+    auto pageCursor = getPageCursorForOffsetInGroup(startOffsetInGroup, state);
     auto numValuesToScan = endOffsetInGroup - startOffsetInGroup;
     scanUnfiltered(
-        transaction, pageCursor, numValuesToScan, resultVector, chunkMeta, offsetInVector);
+        transaction, pageCursor, numValuesToScan, resultVector, state.metadata, offsetInVector);
 }
 
-void Column::scan(node_group_idx_t nodeGroupIdx, ColumnChunk* columnChunk) {
+void BaseColumn::scan(node_group_idx_t nodeGroupIdx, ColumnChunk* columnChunk) {
     if (nullColumn) {
         nullColumn->scan(nodeGroupIdx, columnChunk->getNullChunk());
     }
     if (nodeGroupIdx >= metadataDA->getNumElements()) {
         columnChunk->setNumValues(0);
     } else {
         auto chunkMetadata = metadataDA->get(nodeGroupIdx, TransactionType::WRITE);
+        // Make sure that the chunk is large enough
```

Comment:
hmm. I wonder if we should simply ASSERT here, and the caller should be responsible to pass in a properly sized columnChunk. My concern is this may cause implicit `resize` here without notice.